frontend/src/metabase-lib/filter.ts

```diff
@@ -323,13 +323,19 @@ export function relativeDateFilterClause({
   offsetBucket,
   options,
 }: RelativeDateFilterParts): ExpressionClause {
+  const columnWithoutBucket = withTemporalBucket(column, null);
```

Comment:
Bug 1 - not clearing temporal bucket. If you create an exclude date filter (sets bucket) and then change it to a relative filter, it will not work correctly. There is a unit test for this.